### DISCUSSION TITLE
faro-web-sdk on Chrome extension app

### DISCUSSION THREAD

**Discussion by marcelojfsilva** [grafana/faro-web-sdk#220] [Q&A]
I'm trying to implement the faro-web-sdk in the extension version of chrome. Although, Chrome extension app uses service worker to work the resources in the background.

When I add the faro-web-sdk to the project and run it, I get this error: "Uncaught ReferenceError: window is notdefined".

This error occurs because the object's window does not exist on service worker context. How can i solve this problem to implement Faro on chrome extension app?

**Comment by codecapitano** [comment]
Hi @marcelojfsilva 

Faro should work with Service Workers on the web out of the box[1], but Chrome service workers are a bit different to their web counterparts[2].

One difference is that they can't access the DOM or the window interface[3].

* [1] https://github.com/grafana/faro-web-sdk/pull/195
* [2] https://developer.chrome.com/docs/extensions/mv3/service_workers/
* [3] https://developer.chrome.com/docs/extensions/migrating/to-service-workers/#changes-youll-need-to-make

Cheers,
Marco
